src/System.ObjectModel/src/System/Collections/ObjectModel/ObservableCollection.cs

```diff
@@ -356,13 +364,14 @@ public void Dispose()
 
         #region Private Fields
 
-        private const string CountString = "Count";
-
-        // This must agree with Binding.IndexerName.  It is declared separately
-        // here so as to avoid a dependency on PresentationFramework.dll.
-        private const string IndexerName = "Item[]";
-
-        private SimpleMonitor _monitor = new SimpleMonitor();
+        private readonly SimpleMonitor _monitor = new SimpleMonitor();
         #endregion Private Fields
     }
+
+    internal static class EventArgsCache
```

Comment:
If they were defined on `ObservableCollection<T>`, wouldn't there be multiple instantiations per `T`?

Or are you wondering if they should be on the `PropertyChangedEventArgs` and `NotifyCollectionChangedEventArgs` themselves? Something like:

```c#
public class PropertyChangedEventArgs : EventArgs
{
    internal static readonly PropertyChangedEventArgs Count = new PropertyChangedEventArgs("Count");
    internal static readonly PropertyChangedEventArgs Indexer = new PropertyChangedEventArgs("Item[]");

    ...
}

public class NotifyCollectionChangedEventArgs : EventArgs
{
    internal static readonly NotifyCollectionChangedEventArgs Reset = new NotifyCollectionChangedEventArgs(NotifyCollectionChangedAction.Reset);

    ...
}
```